lasagne/tests/layers/test_local.py

```diff
@@ -0,0 +1,153 @@
+                            if channelwise:
+                                W = np.random.random(
+                                    (num_filters,) + (filter_size,) * 2 +
+                                    input_shape[2:])
+                                output = channelwise_locally_connected2d(
+                                    input, W, flip_filters=flip_filters)
+                            else:
+                                W = np.random.random(
+                                    (num_filters, input_shape[1]) +
+                                    (filter_size,) * 2 + input_shape[2:])
+                                output = locally_connected2d(
+                                    input, W, flip_filters=flip_filters)
+                            yield _convert(input, W, output,
+                                           {'num_filters': num_filters,
+                                            'filter_size': filter_size,
+                                            'flip_filters': flip_filters,
+                                            'channelwise': channelwise})
+
+
+@pytest.fixture
+def DummyInputLayer():
+    def factory(shape):
+        from lasagne.layers.input import InputLayer
+        return InputLayer(shape)
+    return factory
+
+
+class TestLocallyConnected2DLayer:
+    @pytest.mark.parametrize(
+        "input, W, output, kwargs", list(locally_connected2d_test_sets()))
+    def test_defaults(self, DummyInputLayer, input, W, output, kwargs):
+        from lasagne.layers import LocallyConnected2DLayer
+        b, c, h, w = input.shape
+        input_layer = DummyInputLayer((b, c, h, w))
+        layer = LocallyConnected2DLayer(
+                input_layer,
+                W=W,
+                **kwargs)
+        actual = layer.get_output_for(theano.shared(input)).eval()
+        assert actual.shape == output.shape
+        assert actual.shape == layer.output_shape
+        assert np.allclose(actual, output)
+
+    def test_unsupported_settings(self, DummyInputLayer):
+        from lasagne.layers import LocallyConnected2DLayer
+        input_layer = DummyInputLayer((10, 2, 4, 4))
+        for pad in 'valid', 'full', 1:
+            with pytest.raises(NotImplementedError) as exc:
+                LocallyConnected2DLayer(input_layer, 2, 3, pad=pad)
+            assert "requires pad='same'" in exc.value.args[0]
+        with pytest.raises(NotImplementedError) as exc:
+            LocallyConnected2DLayer(input_layer, 2, 3, stride=2)
+        assert "requires stride=1 / (1, 1)" in exc.value.args[0]
+
+    def test_invalid_settings(self, DummyInputLayer):
+        from lasagne.layers import LocallyConnected2DLayer
+        input_layer = DummyInputLayer((10, 2, 4, 4))
+        with pytest.raises(ValueError) as exc:
+            LocallyConnected2DLayer(input_layer, 4, 3, channelwise=True)
+        assert "num_filters and the number of input channels should match"\
+               in exc.value.args[0]
```

Comment:
Thanks for your quick update! You'll need to add a test with `DummyInputLayer((10, None, 4, 4))` or something like this to trigger the new check for a fixed input size.